pallets/liquidity-provider/src/lib.rs

```diff
@@ -56,27 +55,21 @@ decl_module! {
         #[weight = 0]
         pub fn attest(
             origin,
-            collateral: Collateral,
+            asset: Asset,
             balance: Balance<T>
         ) -> dispatch::DispatchResult
         {
-            let who = ensure_signed(origin)?;
-
-            pallet_membership::Module::<T, MintMembers>::members()
-                .binary_search(&who)
-                .ok()
-                .ok_or(pallet_membership::Error::<T, MintMembers>::NotMember)?;
-
-            pallet_membership::Module::<T, ProviderMembers>::members()
-                .binary_search(&who)
-                .ok()
-                .ok_or(pallet_membership::Error::<T, ProviderMembers>::NotMember)?;
-
-            T::Collateral::deposit(collateral.into(), &who, balance)?;
-            T::Collateral::reserve(collateral.into(), &who, balance)?;
-            T::Asset::deposit(Asset::Usdv, &who, balance)?;
-            Self::deposit_event(RawEvent::Attestation(who, collateral));
-            Ok(())
+            match asset {
+                Asset::Usdv => return Err(crate::Error::<T>::MustNotBeUsdv.into()),
+                Asset::Collateral(collateral) => {
+                    let who = ensure_signed(origin)?;
+                    do_attest::<T>(who.clone(), Asset::Usdv, balance)?;
+                    T::Collateral::deposit(collateral.into(), &who, balance)?;
+                    T::Collateral::reserve(collateral.into(), &who, balance)?;
+                    Self::deposit_event(RawEvent::Attestation(who, collateral.into()));
+                    Ok(())
+                }
```

Comment:
Since every attestation is also a mint, everything that is not a stable collateral will fail